### ISSUE TITLE
Feature Request: Allow manual residue position masking in intervene_on_sequences (bypass attribution step)

### ISSUE THREAD

**Issue by etherealsunshine** [prescient-design/lobster#182]
## Description

Currently, `intervene_on_sequences` in the concept bottleneck models always computes a gradient-based attribution map to decide which token positions to mask before applying the concept intervention.

While this is useful for automated concept-based edits, there’s no option to directly specify which residues/tokens to mask for targeted interventions. This makes it cumbersome to make manual edits (e.g., based on prior knowledge, experimental constraints, or targeted mutagenesis).

## Current behavior

```py
# Always does gradient-based attribution:
attribution = torch.autograd.grad(...)
mask = self._create_intervene_mask(attribution, ...)
```
- No public parameter to override mask with user-defined positions.
- Requires forward+backward passes even if user just wants fixed positions masked.

## Proposed change
Add an optional `manual_positions` parameter to intervene_on_sequences:

## Behavior:

If manual_positions is None → keep current attribution-based masking.

If manual_positions is provided:

- Skip attribution computation entirely.
- Create a (B, L) boolean mask with True at those positions for all sequences in the batch.
- Pass that mask to `._mask_inputs`.

## Example

```py
if manual_positions is None:
    # existing gradient attribution path...
    ...
else:
    mask = torch.zeros_like(input_ids, dtype=torch.bool)
    for pos in manual_positions:
        mask[:, pos] = True

    forward_output = self.model(input_ids=input_ids, inference=True, attention_mask=attention_mask)
    pred_concepts_value = forward_output["concepts"]
```
## Benefits

- Enables precise, reproducible interventions on known residues/tokens.
- Avoids unnecessary gradient computation when positions are pre-determined.
- Broadens usability for structure-guided or experimental-design–driven sequence edits.
- Keeps compatibility with current attribution-based workflow.

**Comment by etherealsunshine** [comment]
I'd be happy to send a PR for this if this approach looks good :)

**Comment by ncfrey** [comment]
@etherealsunshine this approach looks great - please feel free to open a PR!
